File: python/class programs/factorial using function.py
def numinput():
    a=int(input("enter the number:"))
    return a

def factorial(a):
    fac=1
    for i in range(1,a+1):
        fac=fac*i
    return fac

a=numinput()
b=factorial(a)
print(b)

File: c programming/class program but in c/factorial_using_function.c
#include <stdio.h>

int numinput(){
    int a;
    printf("Enter the number:");
    scanf("%d",&a);
    return a;
}

int factorial(int a){
    int f=1;
    for (int i=1;i<a+1;i++){
        f=f*i;
    }
    return f;
}

int main(){
    int a,f;
    a=numinput();
    f=factorial(a);
    printf("The factorial of %d is %d",a,f);
    return 0;
}